src/plugins/intel_gpu/src/graph/include/non_zero_inst.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+#pragma once
+#include "intel_gpu/primitives/non_zero.hpp"
+#include "primitive_inst.h"
+#include "intel_gpu/runtime/error_handler.hpp"
+
+#include <string>
+#include <memory>
+
+namespace cldnn {
+
+template <>
+struct typed_program_node<count_nonzero> : public typed_program_node_base<count_nonzero> {
+    using parent = typed_program_node_base<count_nonzero>;
+    typed_program_node(const std::shared_ptr<count_nonzero> prim, program& prog) : parent(prim, prog) {
+        support_padding_all(true);
+    }
+public:
+    using parent::parent;
+
+    program_node& input() const {
+        CLDNN_ERROR_LESS_THAN(id(), "the number of dependencies", dependencies.size(), "1", 1, "ERROR: the node has no input");
+        return get_dependency(0);
+    }
+};
+
+using count_nonzero_node = typed_program_node<count_nonzero>;
+
+template <>
+class typed_primitive_inst<count_nonzero> : public typed_primitive_inst_base<count_nonzero> {
+    using parent = typed_primitive_inst_base<count_nonzero>;
+
+public:
+    static layout calc_output_layout(count_nonzero_node const& node);
+    static std::string to_string(count_nonzero_node const& node);
+
+    typed_primitive_inst(network& network, count_nonzero_node const& node);
+
+private:
+    void on_execute() override;
+};
+
+using count_nonzero_inst = typed_primitive_inst<count_nonzero>;
+
+template <>
+struct typed_program_node<gather_nonzero> : public typed_program_node_base<gather_nonzero> {
+    using parent = typed_program_node_base<gather_nonzero>;
+    typed_program_node(const std::shared_ptr<gather_nonzero> prim, program& prog) : parent(prim, prog) {
+        support_padding_all(true);
+    }
+
+    mutable bool _shape_ready = false;
+public:
+    using parent::parent;
+
+    program_node& input(size_t index = 0) const {
+        CLDNN_ERROR_LESS_THAN(id(), "the number of dependencies", dependencies.size(), "1", 1, "ERROR: the node has no input");
+        return get_dependency(index);
+    }
+
+    ov::PartialShape output_shape = ov::PartialShape({ov::Dimension::dynamic(), ov::Dimension::dynamic(), 1, 1});
```

Comment:
I think we dont need this output_shae variable even in the program_node, because it has output_layout. We only need to update the program_node::output_layout at update_shape, which is initialized here.

Reply:
I removed the `output_shape` and replaced its uses with `output_layout`. Thank you!